src/reason_parser.mly

```diff
@@ -3079,31 +3079,22 @@ _or_pattern:
   | pattern BAR pattern
     { mkpat(Ppat_or($1, $3)) }
 
-
-/**
- * Makes the bar part of the pattern location. This makes it much easier to
- * interleave comments. Normally in switch/fun| each case doesn't have a
- * position, but the left end of the pattern and right end of the expression is
- * used as the location. We need to make the left location of the pattern match
- * what end of line comments appear to be attaching to.
- */
-bar_located_pattern: BAR pattern  {
```

Comment:
Yes that's true. 

For single "|" case, we already have a SourceMap that includes the bar, so we are able to attach to the entire match case.